parser_library/src/analyzer.cpp

```diff
@@ -67,13 +69,36 @@ std::unique_ptr<processing::preprocessor> analyzer_options::get_preprocessor(
 
     struct combined_preprocessor : processing::preprocessor
     {
+        mutable std::vector<std::unique_ptr<semantics::preprocessor_statement_si>> m_statements;
```

Comment:
I definitely don't like the `mutable` here and given the relevant stuff is held in the unique_ptr I don't really see the need for it.
Maybe `get_statements` should just return a value?